### PR TITLE
Modal keybindings

### PR DESCRIPTION
**Pull request by MrStashley**
This currently doesn't work 

For some reason my ModalKeybindings class doesn't mount when a footer is on the screen, and I'm not sure why

### CODE REVIEW

File: src/app/common/elements/modal.tsx [hunk 1]
```diff
@@ -32,8 +34,39 @@ interface ModalFooterProps {
...
+    }
+
+    render(): React.ReactNode {
+        return null;
+    }
+}
+
 const ModalFooter: React.FC<ModalFooterProps> = ({ onCancel, onOk, cancelLabel = "Cancel", okLabel = "Ok" }) => (
     <div className="wave-modal-footer">
+        <ModalKeybindings onOk={onOk} onCancel={onCancel}></ModalKeybindings>
```
Comment (MrStashley) [inline src/app/common/elements/modal.tsx:69]: modal keybindings class here

componentDidMount doesn't get called, even though it seems like this class should be on the screen

File: src/app/common/elements/modal.tsx [hunk 2]
```diff
@@ -30,10 +33,53 @@ interface ModalFooterProps {
...
+
+const ModalFooter: React.FC<ModalFooterProps> = ({
+    onCancel,
+    onOk,
+    cancelLabel = "Cancel",
+    okLabel = "Ok",
+    keybindings = true,
+}) => (
     <div className="wave-modal-footer">
+        <If condition={keybindings}>
```
Comment (MrStashley) [inline src/app/common/elements/modal.tsx:80]: I kept keybindings as part of the footer but added an optional switch for it that's defaulted to true

The unusual footers use the class directly, but the modal footer has the keybindings built in still
Is this too magic? I figured it was convenient but I can get rid of it

File: src/app/common/elements/inlinesettingstextedit.tsx [hunk 3]
```diff
@@ -60,24 +68,40 @@ class InlineSettingsTextEdit extends React.Component<
...
+        });
+    }
+
+    unregisterKeybindings() {
+        let domain = "inline-settings" + this.curId;
+        GlobalModel.keybindManager.unregisterDomain(domain);
+    }
+
+    handleBlur() {
+        console.log("handle blur");
```
Comment (sawka) [inline src/app/common/elements/inlinesettingstextedit.tsx:97]: remove console.log

File: src/app/common/elements/inlinesettingstextedit.tsx [hunk 4]
```diff
@@ -60,24 +68,40 @@ class InlineSettingsTextEdit extends React.Component<
...
+        this.unregisterKeybindings();
+        this.cancelChange();
+    }
+
+    componentWillUnmount(): void {
+        console.log("handle unmoount");
```
Comment (sawka) [inline src/app/common/elements/inlinesettingstextedit.tsx:103]: move console.log

**Pull request merged**